### PR TITLE
Hw selenide

### PR DESCRIPTION
**Pull request by Anastasiia-Bespalko**

### CODE REVIEW

File: src/test/java/Apteka911/Test1Selenide.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+import com.codeborne.selenide.WebDriverRunner;
+import com.github.javafaker.Faker;
+
+import java.time.Duration;
+
+import static com.codeborne.selenide.Condition.visible;
+import static com.codeborne.selenide.Selenide.$;
+import static com.codeborne.selenide.Selenide.$x;
+
+public class Test1Selenide {
```
Comment (AntBelyi) [inline src/test/java/Apteka911/Test1Selenide.java]: Bad name for test class. It should be logical. Maybe about what test verify

File: src/test/java/Apteka911/RegistrationTest1Selenide.java [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+        String email = faker.internet().emailAddress();
+        String password = faker.internet().password();
+
+        SelenideElement input = $x("//a[@class='personal-area__user-name']");
+        input.click();
+
+        SelenideElement registration = $x("//a[@class='personal-area__register']").shouldBe(visible);
+        registration.click();
+
+        SelenideElement name = $x("//input[@type='text'][@name='userFirstname']").should(Condition.appear, Duration.ofSeconds(10));
```
Comment (AntBelyi) [inline src/test/java/Apteka911/RegistrationTest1Selenide.java]: "Duration.ofSeconds(10)" can be removed from "should()" method. Remove please from all methods from all classes

File: src/test/java/Apteka911/RegistrationTest1Selenide.java [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+        SelenideElement password1 = $x("//input[@type='password'][@name='password1']");
+        password1.should(Condition.appear, Duration.ofSeconds(10)).setValue(password);
+
+        $(".btn.btn-accent.btn-m.btn_mw-180").click();
+
+        $("p.alert.alert-error").should(Condition.appear, Duration.ofSeconds(10));
+
+        String expectedErrorMessage = "Не всі обов'язкові поля заповнені або помилка введення даних";
+        String actualErrorMessage = $("p.alert.alert-error").text();
+        assert actualErrorMessage.contains(expectedErrorMessage) : "Сообщение об ошибке не соответствует ожидаемому.";
```
Comment (AntBelyi) [inline src/test/java/Apteka911/RegistrationTest1Selenide.java]: change assert

File: src/test/java/Apteka911/RegistrationTest1Selenide.java [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+        email1.should(Condition.appear, Duration.ofSeconds(10)).setValue(email);
+
+        SelenideElement password1 = $x("//input[@type='password'][@name='password1']");
+        password1.should(Condition.appear, Duration.ofSeconds(10)).setValue(password);
+
+        $(".btn.btn-accent.btn-m.btn_mw-180").click();
+
+        $("p.alert.alert-error").should(Condition.appear, Duration.ofSeconds(10));
+
+        String expectedErrorMessage = "Не всі обов'язкові поля заповнені або помилка введення даних";
```
Comment (AntBelyi) [inline src/test/java/Apteka911/RegistrationTest1Selenide.java]: move this "expectedErrorMessage" to the very beginning of "main" method

File: src/test/java/Apteka911/RegistrationTest1Selenide.java [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+        SelenideElement lastName1 = $x("//input[@type='text'][@name='userLastname']");
+        lastName1.should(Condition.appear, Duration.ofSeconds(10)).setValue(lastName);
+
+        SelenideElement phone = $x("//input[@name='userPhone']");
+        phone.should(Condition.appear, Duration.ofSeconds(10)).setValue(phoneNumber);
+
+        SelenideElement email1 = $x("//input[@type='text'][@name='userEmail']");
+        email1.should(Condition.appear, Duration.ofSeconds(10)).setValue(email);
+
+        SelenideElement password1 = $x("//input[@type='password'][@name='password1']");
```
Comment (AntBelyi) [inline src/test/java/Apteka911/RegistrationTest1Selenide.java]: "password1" its a bad name! what mean 1? rename into passwordElement for example. And change please for all elements with 1 number in name

File: src/test/java/Apteka911/ProductCatalogTest4Selenide.java [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+public class ProductCatalogTest4Selenide {
+    public static void main(String[] args) {
+        Selenide.open("https://apteka911.ua/ua");
+        WebDriverRunner.getWebDriver().manage().window().maximize();
+
+        SelenideElement iconMenu = $x("//i[@class='icon-menu']");
+        iconMenu.click();
+
+        SelenideElement medicinalProducts = $x("//*[@id=\"wrap\"]/div[3]/div/div/div[1]/ul/li[4]/a");
+        medicinalProducts.hover();
```
Comment (AntBelyi) [inline src/test/java/Apteka911/ProductCatalogTest4Selenide.java]: Why do you use this "hover()" method? If you want to check that element appears after hovering then you need to change xpath from 24 line.

File: src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java [hunk 7]
```diff
@@ -0,0 +1,39 @@
...
+        Selenide.open("https://apteka911.ua/ua");
+        WebDriverRunner.getWebDriver().manage().window().maximize();
+
+        SelenideElement input = Selenide.$x("//a[@class='personal-area__user-name']");
+        input.click();
+
+        SelenideElement login = Selenide.$x("//a[@class='personal-area__user-login']");
+        login.click();
+
+        SelenideElement emainOrNumber = Selenide.$x("//input[@type='text'][@name='login']");
```
Comment (AntBelyi) [inline src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java]: mistake in variable name

File: src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+        password.setValue("password");
+
+        SelenideElement entryButton = Selenide.$x("//input[@class='btn btn-accent btn-m btn_mw-180']");
+        entryButton.click();
+
+        SelenideElement successfulLogin = Selenide.$x("//a[@class='personal-area__user-name']");
+        successfulLogin.shouldBe(Condition.visible);
+
+        SelenideElement userLink = $x("//a[@class='personal-area__user-name']");
+        assert userLink.is(Condition.visible) : "Элемент не отображается после входа";
```
Comment (AntBelyi) [inline src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java]: assert should be changed

File: src/test/java/Apteka911/MedicationSearchTest3Selenide.java [hunk 9]
```diff
@@ -0,0 +1,29 @@
...
+        search.click();
+
+        SelenideElement textInput = $x("//input[@id='search']");
+        textInput.sendKeys("спазмалгон");
+
+        SelenideElement buttonSearch = $x("//button[@class='b-search__button-icon icon-search']");
+        buttonSearch.click();
+
+        SelenideElement searchResult = $x("//h1[contains(text(),'Спазмалгон')]");
+        String expectedText = "Спазмалгон";
```
Comment (AntBelyi) [inline src/test/java/Apteka911/MedicationSearchTest3Selenide.java]: where this variable is used?

File: src/test/java/Apteka911/Test1Selenide.java (hunk 1 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/Test1Selenide.java]: fixed

File: src/test/java/Apteka911/RegistrationTest1Selenide.java (hunk 2 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/RegistrationTest1Selenide.java]: fixed

File: src/test/java/Apteka911/RegistrationTest1Selenide.java (hunk 3 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/RegistrationTest1Selenide.java]: fixed

File: src/test/java/Apteka911/RegistrationTest1Selenide.java (hunk 4 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/RegistrationTest1Selenide.java]: fixed

File: src/test/java/Apteka911/RegistrationTest1Selenide.java (hunk 5 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/RegistrationTest1Selenide.java]: fixed

File: src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java (hunk 7 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java]: fixed

File: src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java (hunk 8 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/PersonalAccountLoginTest2Selenide.java]: fixed

File: src/test/java/Apteka911/MedicationSearchTest3Selenide.java (hunk 9 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/MedicationSearchTest3Selenide.java]: fixed

File: src/test/java/Apteka911/RegistrationTest1Selenide.java (hunk 2 above)
Comment (AntBelyi) [reply src/test/java/Apteka911/RegistrationTest1Selenide.java]: Not fixed. Check my comment more attentive and make ALL fixes

File: src/test/java/Apteka911/MedicationSearchTest3Selenide.java [hunk 10]
```diff
@@ -0,0 +1,30 @@
...
+public class MedicationSearchTest3Selenide {
+    public static void main(String[] args) {
+        Selenide.open("https://apteka911.ua/ua");
+        WebDriverRunner.getWebDriver().manage().window().maximize();
+
+        SelenideElement search = $x("//input[@id='search']");
+        search.click();
+
+        SelenideElement textInput = $x("//input[@id='search']");
+        textInput.sendKeys("спазмалгон");
```
Comment (AntBelyi) [inline src/test/java/Apteka911/MedicationSearchTest3Selenide.java]: "спазмолгон" is used in several places. Please move it to the separate varibale in the very beginning of the main method

File: src/test/java/Apteka911/RegistrationTest1Selenide.java (hunk 2 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/RegistrationTest1Selenide.java]: fixed

File: src/test/java/Apteka911/ProductCatalogTest4Selenide.java (hunk 6 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/ProductCatalogTest4Selenide.java]: fixed

File: src/test/java/Apteka911/MedicationSearchTest3Selenide.java (hunk 10 above)
Comment (Anastasiia-Bespalko) [reply src/test/java/Apteka911/MedicationSearchTest3Selenide.java]: fixed

**Pull request merged**